users/migrations/0005_transfer_to_pysocial.py

```diff
@@ -0,0 +1,58 @@
+from django.db import connection, migrations
+from django.db.utils import ProgrammingError
+
+
+def transfer_users_from_allauth_to_pysocial(apps, schema_editor):
+    """
+    Transfer user UIDs from Allauth to Python Social Auth.
+    """
+    all_tables = connection.introspection.table_names()
+    if "socialaccount_socialaccount" in all_tables:
+        # allauth
+        with connection.cursor() as cursor:
+            cursor.execute(
+                "SELECT user_id, uid, extra_data FROM socialaccount_socialaccount"
+            )
+            social_accounts = cursor.fetchall()
+
+        # pysocial
+        UserSocialAuth = apps.get_model("social_django", "UserSocialAuth")
+
+        social_auths = []
+        for user_id, uid, extra_data in social_accounts:
+            social_auths.append(
+                UserSocialAuth(
+                    user_id=user_id,
+                    provider="tunnistamo",  # Defined in TunnistamoOIDCAuth
+                    uid=uid,
+                    extra_data=extra_data,
+                )
+            )
+        UserSocialAuth.objects.bulk_create(social_auths, ignore_conflicts=True)
```

Comment:
If the amount of users is less than 1000 this could be OK, but for much larger sets, e.g. 100000 users it's probably not a good idea to create them in one bunch. If you know that there's not so many users to migrate, this (nor [the other place](https://github.com/City-of-Helsinki/helerm/pull/308/commits/043b74eb58d83eee44bb2e1f0640ccfef1b902ab#r581756149)) won't need any changes. Otherwise it should be changed to iterate them in batches (e.g. 100 at a time).

Reply:
Or maybe you could use raw SQL for the creation too. That way you wouldn't need to pull any user data to Python at all.
```sql
INSERT INTO {dest_table} (user_id, provider, uid, extra_data)
 SELECT user_id, 'tunnistamo', uid, extra_data FROM {src_table};
```

Reply:
Fixed. Had to add the current timestamps there as well because it requires the created and modified dates.